### DISCUSSION TITLE
How To Make a project's assets download from start?

### DISCUSSION THREAD

**Discussion by Mashimaro7** [codemagic-ci-cd/codemagic-docs#1962] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Unity

### Steps to reproduce

1. Get Codemagic set up
2. Download project's assets at start scene
3. Start game, have fun.

### Expected results

Profit

### Actual results

I don't know how to do this? I read this article, 
https://blog.codemagic.io/unity-cloud-content-delivery/
But it seems to be more about automating builds rather than downloading all assets on start? Basically i want the start scene of my game to check the server and see if there are any changes in the CCD or Github repo(not sure which one?) and i have no idea how to set this up. 

There's two reasons i want this set up in this way, 1. I want the build size to be smaller and 2. I want the game to automatically update on all platforms without the user having to manually build. I have builds for stand alone Windows and mac, and am planning to get it released for Windows and mac app stores(PC and Apple, not mobile) once i can get some issues worked out. 

Am i even looking in the right place? I mean, i know how to use asset bundles to get this sorted out, but that would require me to rewrite a huge chunk of my project, get everything set up in some manager script of sorts and manually assign everything, it's a very large project, so i don't think this would be as simple as taking the whole project and putting it into an asset bundle and updating it whenever i change something(or maybe it is that easy? haha)

To reiterate, my goal is to have most assets of the project not included in the build, have an initial download scene that checks if the files are outdated, and downloads all the ones that are, and starts the game when fully updated. If anyone could give me instructions show me a guide on how to do this with Codemagic, or tell me the best way to go about this that'd be great. I'm kinda at my wits end haha. I looked through the docs but can't really find what i'm looking for, is this even what Codemagic is for?

**Comment by icarusdust** [comment]
Hi @Mashimaro7, when starting a build, your project needs to be cloned and as this process happens in a virtual environment, the environment needs to be re-created with each build trigger. So, in short your repo will have to be cloned, and by default if your assets are stored in the repo, they all will be cloned as well. 

You can configure automatic build triggering and add a condition to skip builds in case of changes in a particular path, but I guess you still want it to complete a build, not skipping it but rather do not download all the assets (only the ones that are updated or newly added) I believe, a good start would be storing your assets in a 3rd party storage and download them as soon as your project cloning is complete. In that case, you can add a custom conditional script to skip downloading all the assets. 

Other than that, [skipping builds](https://docs.codemagic.io/yaml-running-builds/starting-builds-automatically/#using-when-to-run-or-skip-builds) on certain events unless there are changes in the assets directory could be your approach. However, please note that this approach is not for downloading certain assets, it will be either skipping builds or downloading them all.

**Reply by icarusdust** [reply]
This 2 approach is what comes to my mind at the moment. I think, ideally you should figure out how you would do it when downloading assets on your local machine. Then, whatever condition you are setting will work the same way with Codemagic builds.

**Comment by icarusdust** [comment]
I am closing the thread due to inactivity, you can re-open it by replying to it.
